bluetooth: stm32wba: add temperature calibration of the linklayer

rjayles opened the pull request:
This feature add the ability for the linklayer to calibrate itself using the current die temperature.
This feature can be enabled/disabled using a dedicated config.

rjayles committed:
drivers: bluetooth: dedicated Kconfig for the STM32WBA configuration

Adds a Kconfig for the STM32WBA associated BT configuration

The first configuration added is the possibility to activate the
calibration of the radio using the temperature

Signed-off-by: Romain Jayles <romain.jayles@st.com>

rjayles committed:
bluetooth: stm32wbax: add temperature calibration of linklayer

This patch allows to link the request of the linklayer for a
temperature calibration to the temperature driver.
The linklayer will then adapt and trigger its calibration related to
the current temperature.

Signed-off-by: Romain Jayles <romain.jayles@st.com>

mathieuchopstm commented on soc/st/stm32/stm32wbax/hci_if/ll_sys_if_adapt.c:
```diff
@@ -38,6 +47,81 @@ void ll_sys_bg_process_init(void)
+ */
+void ll_sys_bg_temperature_measurement_init(void)
+{
+	if (!dev_temp_sensor) {
+		LOG_ERR("dev_temp_sensor is null\n");
+		k_panic();
+	} else {
```
This cannot happen (`DEVICE_DT_GET` causes build errors if device doesn't exist/is not instanced)
```suggestion
```

(even if it *could* happen, it should be checked via a `BUILD_ASSERT` rather than at runtime)

GeorgeCGV commented on soc/st/stm32/stm32wbax/hci_if/ll_sys_if_adapt.c:
```diff
@@ -38,6 +47,76 @@ void ll_sys_bg_process_init(void)
+	int rc = sensor_sample_fetch(dev_temp_sensor);
+
+	if (rc) {
+		LOG_ERR("Failed to fetch sample (%d)\n", rc);
```
Minor: the newline is unnecessary in the log output.

jhedberg reviewed (changes requested):
A few cosmetic issues with Kconfig

mathieuchopstm commented on soc/st/stm32/stm32wbax/hci_if/ll_sys_if_adapt.c:
```diff
@@ -11,11 +11,20 @@
 
 #include "ll_sys.h"
+#include "ll_intf_cmn.h"
+#if (USE_TEMPERATURE_BASED_RADIO_CALIBRATION == 1)
```
Nit: we're on the Zephyr side so we *should* have visibility on the Kconfig symbols?
```suggestion
#if defined(CONFIG_BT_STM32WBA_USE_TEMPERATURE_BASED_RADIO_CALIBRATION)
```
Ditto for all such usage.

mathieuchopstm commented on soc/st/stm32/stm32wbax/hci_if/ll_sys_if_adapt.c:
```diff
@@ -38,6 +47,76 @@ void ll_sys_bg_process_init(void)
+ * @param  None
+ * @retval None
+ */
+static void ll_sys_temperature_calibration_measurement(void)
```
Cosmetic nit: this is private so we don't have to name this `ll_sys_<...>`

Could be renamed to e.g. `perform_temperature_measurement`.

mathieuchopstm reviewed (dismissed):
Otherwise LGTM.

mathieuchopstm reviewed (commented):
Also

erwango commented:
Please rebase.

rjayles committed:
west.yml: references hal_stm32 pull request for temperature calibration

Add the pull request reference #283 of hal_stm32 related to temperature
calibration activation

Signed-off-by: Romain Jayles <romain.jayles@st.com>
